source/common/unified_memory_allocation/include/uma/memory_pool_ops.h

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef UMA_MEMORY_POOL_OPS_H
+#define UMA_MEMORY_POOL_OPS_H 1
+
+#include <uma/base.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// TODO: create macro for intializing this structure (including version fields)
+struct uma_memory_pool_ops_t {
```

Comment:
I've added the comment but I'm not sure about the naming. Right now, we don't use the word "heap" anywhere. If we'd mention it here, we would need to explain what it is exactly and what is the relation between pool and heap. And I think it's not exactly clear because those operations we define here are actually responsible for "pooling" (managing memory from the provider, etc.)

 I think I'd rather rename `pool` to `heap` if you think that makes more sense.

Reply:
Ok, I agree with you regarding naming. let's go with pool.